### PR TITLE
Feature tests

### PR DESCRIPTION
**Pull request by Jesse193**
This refactors the views so that it no longer contains facade method names. This also implements the login and logout methods in the sessions controller. The flash error messages were removed from the application controller do to them not being used. This also adds a test to test if a user is not logged in and attempts to favorite a market. Note: Some markets may throw an error for routes that is due to it not having any. I'm working on an error handling for that. Edit: Removal of to_i and to_f from controller. They were unnecessary

### CODE REVIEW

**Review by kbergstrom78 — APPROVED** [review]
This looks really good, Jesse. Nice work!

File: Gemfile [hunk 1]
```diff
@@ -68,8 +68,8 @@ group :test do
   gem 'webmock'
   gem 'factory_bot_rails'
   gem 'vcr'
-  # gem 'selenium-webdriver'
-  # gem "webdrivers"
+  gem 'selenium-webdriver'
```
Comment (AStilmock) [inline Gemfile:71]: Lesson learned, don't delete the selenium webdriver! Good catch here

File: app/controllers/routes_controller.rb [hunk 2]
```diff
@@ -1,19 +1,5 @@
...
-
-  def show
-    @market = MarketFacade.new(params)
-    location = Geocoder.search(@market.market.address)
-    params[:latitude] = location.first.coordinates[0]
-    params[:longitude] = location.first.coordinates[1]
-    params[:original_lat] = params[:original_lat].to_f
-    params[:original_lon] = params[:original_lon].to_f
-    @route = RouteFacade.new(params)
+    @routes = RouteFacade.new(params).all_routes
```
Comment (AStilmock) [inline app/controllers/routes_controller.rb:3]: good update!

File: spec/features/users/favorites_spec.rb [hunk 3]
```diff
@@ -99,4 +99,43 @@
...
+          "snap_option" => "Accept EBT at a central location;",
+          "accepted_payment" => "Debit card/Credit card;",
+          "longitude" => -71.202747,
+          "latitude" => 43.580593
+        }
+      }
+
+      stub_request(:get, "https://foodhaven-be.onrender.com/api/v1/markets/#{market_id}")
+        .to_return(body: { data: market_data }.to_json, status: 200)
+      visit market_path(market_id)
```
Comment (AStilmock) [inline spec/features/users/favorites_spec.rb:126]: Nice Test!

**Review by AStilmock — APPROVED** [review]
Big updates here thanks for getting these taken care of!

**Pull request merged**